WordPressKitTests/DomainsServiceRemoteRESTTests.swift

```diff
@@ -238,4 +243,37 @@ class DomainsServiceRemoteRESTTests: RemoteTestCase, RESTTestable {
 
         waitForExpectations(timeout: timeout, handler: nil)
     }
+
+    // MARK: - Get All Domains Tests
+
+    func testGetAllDomainsSucceeds() {
+        let expect = expectation(description: "Get All Domains Succeeds")
+
+        stub { request -> Bool in
+            let url = request.url
+            let matched = url?.path == self.allDomainsEndpoint && url?.query == "no_wpcom=true&resolve_status=true&locale=en"
+            XCTAssertTrue(matched)
+            return matched
+        } response: { request in
+            let path = OHPathForFile(self.allDomainsMockFilename, type(of: self)) ?? ""
+            return fixture(filePath: path, status: 200, headers: nil)
+        }
+
+        var params = DomainsServiceRemote.AllDomainsEndpointParams()
+        params.noWPCOM = true
+        params.resolveStatus = true
+        params.locale = "en"
+        remote.fetchAllDomains(params: params) { result in
+            switch result {
+            case .success(let domains):
+                let expectedCount = 10
+                XCTAssertEqual(domains.count, expectedCount, "There should be \(expectedCount) domains returned.")
+            case .failure:
+                XCTFail("Get All Domains request failed")
+            }
+            expect.fulfill()
+        }
+
+        waitForExpectations(timeout: timeout)
+    }
```

Comment:
Did you consider adding a test for the failure path?

Reply:
Also, the core of the logic here is how the parameters translate to the query string. The implementation checks the for a value and only adds it to the query string then. Have you considered testing that in isolation and in more details? With in isolation, I'm thinking a method on `AllDomainsEndpointParams` that spits out the `[URLQueryItem]` (or `[String: Any]` if we rely on the API client method).

Reply:
> Did you consider adding a test for the failure path?

There are 3 cases where the `fetchAllDomains` could fail:
- The `params` encoding fails.
- The request fails due to networking issue, or the server returns 400 or 500 error. 
- The json decoding fails.

I added new unit tests covering the 1st case 677fc351ab45f8e919b0c313e55269b16e2dd6ca. Not sure we can much about the 2nd case though.

Reply:
There are ways to test both scenarios. I'll see if I can make an example PR about it, maybe next week.